ThingiBrowser/drivers/myminifactory/MyMiniFactoryApiClient.py

```diff
@@ -1,39 +1,77 @@
-# Copyright (c) 2020 Chris ter Beke.
+# Copyright (c) 2020.
 # Thingiverse plugin is released under the terms of the LGPLv3 or higher.
 from typing import List, Callable, Any, Optional, Tuple
+from urllib.parse import urlencode
 
 from PyQt5.QtCore import QUrl
+from PyQt5.QtGui import QDesktopServices
 from PyQt5.QtNetwork import QNetworkRequest, QNetworkReply
 
 from ...Settings import Settings
 from ...PreferencesHelper import PreferencesHelper
 from ...api.ApiHelper import ApiHelper
 from ...api.AbstractApiClient import AbstractApiClient
-from ...api.JsonObject import ApiError, Collection, Thing, ThingFile
+from ...api.JsonObject import ApiError, Collection, Thing, ThingFile, UserData
+from ...api.LocalAuthService import LocalAuthService
 
 
 class MyMiniFactoryApiClient(AbstractApiClient):
     """ Client for interacting with the MyMiniFactory API. """
 
     def __init__(self) -> None:
+        self._username: Optional[str] = None
```

Comment:
Did you mean `self._username = None  # type: Optional[str]`? This line currently causes the plugin to fail while being loaded in Cura.

Reply:
You're right. Cura is using Python 3.5 in the built distribution, but I'm running 3.7 while running Cura from source, so I didn't catch this.